test/unit/CMakeLists.txt

```diff
@@ -20,19 +22,56 @@ include_directories(${EXTRA_CORE_INCLUDE_DIRS})
 
 include(CMakeParseArguments)
 
+message(STATUS "Add test suite unit")
 function(create_unit_test)
   cmake_parse_arguments(
     UNIT
     ""
     "PREFIX"
-    "SOURCES;LIBRARIES;COMPILE_DEFINITIONS"
+    "SOURCES;DEPENDS;LIBRARIES;COMPILE_DEFINITIONS"
     ${ARGN}
   )
-  message(STATUS "Creating unit test ${UNIT_PREFIX}.test")
   add_executable(${UNIT_PREFIX}.test ${UNIT_SOURCES})
   target_compile_definitions(${UNIT_PREFIX}.test PRIVATE ${UNIT_COMPILE_DEFINITIONS})
   target_link_libraries(${UNIT_PREFIX}.test ${UNIT_LIBRARIES})
   set(UNIT_TEST_TARGETS "${UNIT_TEST_TARGETS} ${UNIT_PREFIX}.test" PARENT_SCOPE)
+  # XXX: The call produces both test and target
+  # <${UNIT_PREFIX}.test-deps> as a side effect.
```

Comment:
Fixed:

```diff
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 21e7e36f55..410aab6f54 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -20,27 +20,30 @@ include_directories(${CURL_INCLUDE_DIRS})
 
 include_directories(${EXTRA_CORE_INCLUDE_DIRS})
 
+set(TEST_SUITE_NAME "tarantool-unit")
+
+# XXX: The call produces both test and target
+# <${TEST_SUITE_NAME}.test-deps> as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}.test
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS ${UNIT_TEST_TARGETS}
+)
+
 include(CMakeParseArguments)
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
 
-message(STATUS "Add test suite unit")
 function(create_unit_test)
   cmake_parse_arguments(
     UNIT
     ""
     "PREFIX"
-    "SOURCES;DEPENDS;LIBRARIES;COMPILE_DEFINITIONS"
+    "SOURCES;LIBRARIES;COMPILE_DEFINITIONS"
     ${ARGN}
   )
   add_executable(${UNIT_PREFIX}.test ${UNIT_SOURCES})
   target_compile_definitions(${UNIT_PREFIX}.test PRIVATE ${UNIT_COMPILE_DEFINITIONS})
   target_link_libraries(${UNIT_PREFIX}.test ${UNIT_LIBRARIES})
   set(UNIT_TEST_TARGETS "${UNIT_TEST_TARGETS} ${UNIT_PREFIX}.test" PARENT_SCOPE)
-  # XXX: The call produces both test and target
-  # <${UNIT_PREFIX}.test-deps> as a side effect.
-  _add_test_suite_target(${UNIT_PREFIX}.test
-    LABELS unit
-    DEPENDS ${UNIT_DEPENDS}
-  )
   # Every unit test is a self-contained program without
   # dependencies that does not require a special test environment.
   # Historically unit tests are running using test-run.py because
@@ -70,8 +73,8 @@ function(create_unit_test)
   add_test(NAME ${test_title}
            COMMAND ${test_command})
   set_tests_properties(${test_title} PROPERTIES
-    LABELS unit
-    DEPENDS ${UNIT_PREFIX}.test-deps
+    LABELS ${TEST_SUITE_NAME}
+    DEPENDS ${TEST_SUITE_NAME}.test-deps
     FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
   )
 endfunction()
@@ -447,7 +450,6 @@ create_unit_test(PREFIX luaL_iterator
                            ${ICU_LIBRARIES}
                            ${LUAJIT_LIBRARIES}
                            ${LIB_DL}
-                 DEPENDS unit server coll core misc
 )
 
 create_unit_test(PREFIX say
@@ -467,7 +469,6 @@ set(ITERATOR_TEST_LIBS core tuple xrow unit)
 create_unit_test(PREFIX vy_mem
                  SOURCES vy_mem.c ${ITERATOR_TEST_SOURCES} core_test_utils.c
                  LIBRARIES ${ITERATOR_TEST_LIBS} ${LIB_DL}
-                 DEPENDS ${ITERATOR_TEST_LIBS}
 )
 
 create_unit_test(PREFIX vy_point_lookup
@@ -506,19 +507,16 @@ create_unit_test(PREFIX vy_write_iterator
                          ${ITERATOR_TEST_SOURCES}
                          core_test_utils.c
                  LIBRARIES xlog ${ITERATOR_TEST_LIBS} ${LIB_DL}
-                 DEPENDS xlog ${ITERATOR_TEST_LIBS}
 )
 
 create_unit_test(PREFIX vy_cache
                  SOURCES vy_cache.c ${ITERATOR_TEST_SOURCES} core_test_utils.c
                  LIBRARIES ${ITERATOR_TEST_LIBS} ${LIB_DL}
-                 DEPENDS ${ITERATOR_TEST_LIBS}
 )
 
 create_unit_test(PREFIX coll
                  SOURCES coll.cpp core_test_utils.c
                  LIBRARIES coll unit misc ${LIB_DL}
-                 DEPENDS coll unit misc
 )
 
 create_unit_test(PREFIX tuple_bigref
@@ -535,7 +533,6 @@ create_unit_test(PREFIX checkpoint_schedule
                  SOURCES checkpoint_schedule.c
                          ${PROJECT_SOURCE_DIR}/src/box/checkpoint_schedule.c
                  LIBRARIES m unit
-                 DEPENDS unit
 )
 
 create_unit_test(PREFIX sio
```

The number of test titles in suite `tarantool-unit` is equal to a number of files `.test` in `build/test/unit` + 1.